File: src/Types/order.ts
export type OrderType = 'table' | 'room';

export interface OrderModel {
    fullname : string;
    orderType : OrderType;
    identifier : string | number;
}

export interface Order{
    id: number;
    business_id: number;
    business_name: string;
    order_number: string;
    table_id: number | null;
    table_number?: number | null;
    room_id: number | null;
    is_room_order: boolean;
    guest_id: number | null;
    order_type_id: number;
    order_type_name: string;
    status_id: number;
    status_name: string;
    subtotal: string;
    discount: string;
    tax: string;
    total_amount: string;
    notes: string;
    served_by: string | null;
    items: OrderItem[];
    created_at: string;
    updated_at: string;
}

export interface OrderItem{
    id: number; 
    order_id: number;
    order_number: string;
    food_item_id: number;
    food_item_name: string;
    food_item_available: boolean;
    quantity: number;
    unit_price: string; 
    total_price: string; 
    note: string;
    status_id: number;
    status_name: string;
    created_at: string;
    updated_at: string; 
}

export interface CreateOrderItem{
    food_item_id: number;
    quantity: number;
    status_id: number;
    note?: string;
}

export interface CreateOrderRequest{
    business_id: number;
    table_id?: number | null;
    room_id?: number | null;
    guest_id?: number | null;
    order_type_id: number;
    order_number: string;
    status_id: number;
    subtotal: string;
    tax: string;
    discount: string;
    total_amount: string;
    served_by?: number | null;
    notes?: string;
    is_room_order: boolean
    items: CreateOrderItem[];
}

export interface GetOrderResponse{
    orders: Order[];
}

File: src-tauri/src/models/order.rs
use serde::{Deserialize, Serialize};

#[derive(Serialize, Deserialize, sqlx::FromRow)]
pub struct OrderCreate {
    pub business_id: i32,
    pub order_number: String,
    pub table_id: i32,
    pub guest_id: i32,
    pub order_type_id: i32,
    pub status_id: i32,
    pub subtotal: f32,
    pub tax: f32,
    pub discount: f32,
    pub total_amount: f32,
    pub notes: String,
    pub served_by: i32,
}

#[derive(Serialize, Deserialize, Clone, Debug, sqlx::FromRow)]
pub struct OrderItem {
    pub id: i64,
    pub order_id: i32,
    pub food_item_id: i32,
    pub quantity: i32,
    pub unit_price: f32,
    pub total_price: f32,
    pub status_id: i32,
    pub special_instructions: Option<String>,
}

#[derive(Serialize, Deserialize, Clone, Debug)]
pub struct OrderItemCreate {
    pub order_id: i32,
    pub food_item_id: i32,
    pub quantity: i32,
    pub unit_price: f32,
    pub total_price: f32,
    pub status_id: i32,
    pub special_instructions: Option<String>,
}

#[derive(Serialize, Deserialize)]
pub struct OrderCreateResponse {
    pub message: String,
    pub order: OrderCreate,
}
